src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddAuthorGuestTypeFormSubscriber.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Form\EventSubscriber;
+
+use Sylius\Component\Review\Model\ReviewInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class AddAuthorGuestTypeFormSubscriber implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            FormEvents::PRE_SET_DATA => 'preSetData',
+        ];
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $form = $event->getForm();
+        /** @var ReviewInterface $review */
+        $review = $event->getData();
+        $author = $form->getConfig()->getOption('author');
+
+        if (null === $author && null === $review->getAuthor()) {
```

Comment:
From reading this if statement it would seem that there are two places that an author can retrieved from? I'm not very familiar with the Sylius codebase (doing some code review to pass the time) but could this indicate that some form of abstraction is missing?

The best way I can word this would be; is there a definitive source which tells us who the author is? Or is that this piece of code and I'm just a rambling madman? :P